source/common/common/thread.h

```diff
@@ -187,7 +199,7 @@ struct MainThread {
   }
 
 private:
-  std::thread::id main_thread_id_{std::this_thread::get_id()};
+  std::vector<std::thread::id> main_thread_id_{std::this_thread::get_id()};
```

Comment:
This looks more general than needed. It might be clearer just to have main_thread_id_ and test_thread_id_, and init them specifically.

I'm still not sure actually why we are calling "try" from the test thread though. When does that happen?

Reply:
Because sometimes we construct a customized object(for example some factory or manager) from test thread, which is constructed from server main thread in production. The construction of these objects sometimes call try_with_api_boost, which calls a try.

Reply:
What's try_with_api_boost? I did not find that string in my envoy tree.

I'm still curious where 'try' is called on the test thread. Note that if you actually *do* put `try` in test-code, that's fine. You don't need to use the macro in that case.

I think what must be happening that you've observed is that we are calling production code that's intended for use on the main thread, but we are calling it on the test thread. It might also be feasible to post such calls to the main thread rather than directly calling it. It's also possible that's really hard, in which case maybe you could leave a temporary workaround.

Our actions here can be informed by the nature and count of the current uses.